### ISSUE TITLE
The OSC receiver doesn't properly handle switching between sessions

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1885]
### Describe the bug?

If you have an OSC receiver object that's receiving data and you save it to your inventory then spawned it out in a new session the data will not be received until the other object is destroyed in that session or that session is closed

### To Reproduce

1. have an OSC object that's receiving
2. Go to a new session and spawn it out
3. Observe that the data is not being received in the new session
4. Leave the previous session or destroy the OSC object in the original session
5. In the new session you will observe but the data is being received properly

### Expected behavior

The OSC receiver should recognize that the user is not in the present session and the receiver that's currently present with the user should be prioritized

### Screenshots

https://youtu.be/_rOAiCRbyj4

### Resonite Version Number

2024.4.30.495

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

N/A

### Reporters

@epicEaston197

**Comment by shadowpanther** [comment]
If I understand this correctly (going by description only), the issue here is that:
1. The OSC component doesn't respect the focused state of the user it belongs to (might be intentional).
2. OSC doesn't allow multiple connections, and as soon as the first connection is closed the second one connects.

**Comment by ultrawidegamer** [comment]
yes, this sounds like something that could be intentional as the user could disable the connection within the item when focusing/defocusing, but there could also be a configuration added that allows the connection to auto disconnect on defocus and auto reconnect on focus

**Comment by epicEaston197** [comment]
> yes, this sounds like something that could be intentional as the user could disable the connection within the item when focusing/defocusing, but there could also be a configuration added that allows the connection to auto disconnect on defocus and auto reconnect on focus

that should not be a configuration it should be by default because I was unaware of this behavior and I was really annoyed that my thing wasn't working

**Comment by Frooxius** [comment]
We need logs for this.

**Comment by epicEaston197** [comment]
I've added a log file and a demonstration video in the screenshots section

> We need logs for this.

I've added a log file and a demonstration video in the screenshots section

**Comment by epicEaston197** [comment]
> We need logs for this.

I've added a log file and a demonstration video in the screenshots section

**Comment by epicEaston197** [comment]
okay so disabling the OSC receiver when you're not present in the world with flux fixes the issue but this is of course a workaround

**Comment by EmergencyTemporalShift** [comment]
If OSC connection is left connected in other worlds, it shouldn't be exclusive.

**Comment by shadowpanther** [comment]
OSC Receiver is a server though, and listens for incoming connections. Only one process can listen to any port.
So you either disable the receiver when you unfocus (your HRM stops showing when unfocused) or use different ports for different sessions (your Sender would need multiple connections to control multiple Receivers).

Even in the same session, spawning another avatar with the same setup would cause the conflict because ports are the same.

Thus, I'd stick to WS for HRM because it works the other way and the source is the singular server.

—-

What could be done on Resonite side though, is Receiver multiplexing. I.e., if two receivers for the same user use the same port, they both get the same data and use one listener underneath. This would allow controlling multiple copies of the item both in one session and in different sessions.

**Comment by shiftyscales** [comment]
Seeking input from @Frooxius - requested additional information including a log file has been provided to this issue.

**Comment by FlameSoulis** [comment]
Going to have to necro bump this. I am finally diving in and assumed the server receiver was the client itself, not the component. This screws with my hope of converting my websocket CSV nightmares into something easier to work with, but that point gets lost if only connection per Resonite client is allowed.

**Comment by Frooxius** [comment]
This is now fixed in 2026.4.9.1303!

Resonite will use a single instance in the background, so they will no longer conflict.

Thanks!
